torch/_inductor/codegen/cuda/cuda_template.py

```diff
@@ -78,26 +102,16 @@ def generate(self, **kwargs) -> CUDATemplateCaller:
             extra_args=extra_args,
             source_code=code,
         )
-
-        def make_kernel_render(output_node):
-            kernel = CUDATemplateKernel(
-                kernel_name="KERNEL_NAME",
-            )
-            render = functools.partial(
-                self.render,
-                kernel=kernel,
-                output_node=output_node,
-                **kwargs,
-            )
-            return kernel, render
-
+        cuda_template_buffer = CUDATemplateBuffer(
+            template=self,
+            op=op,
+            workspace_size=lambda: bmreq.workspace_size,  # Not known yet, determined via CUDABenchmarkRequest
```

Comment:
Is it necessary to use a lambda function for `workspace_size` here (since we don't support non-zero workspace size anyways)?

Reply:
Not at the moment, as workspace size always has to be zero. I was trying to structure the code such that it won't break once that limitation is lifted. At this point, the workspace size requirement is not fixed yet, since the autotuning didn't run yet. So, we can replace this with a constant 0 if we want to for now.